cdhweb/people/models.py

```diff
@@ -371,70 +381,115 @@ def current_title(self):
+def init_person_from_ldap(user, ldapinfo):
+    '''Extra User init logic for creating and auto-populating a corresponding
+    Person with data from LDAP.'''
 
+    # update User to use ldap email
     user.email = user.email.lower()
     user.save()
 
-    try:
-        profile = user.profile
-    except ObjectDoesNotExist:
-        profile = Profile.objects.create(user=user)
-        # set profiles to draft by default when creating a new profile *only
-        # so we don't get a new page for every account we initialize
-        profile.status = CONTENT_STATUS_DRAFT
-
-    # populate profile with data we can pull from ldap
-    # but do not set any fields with content, which may contain edits
-
-    # - set user's display name as page title
-    if not profile.title:
-        profile.title = str(ldapinfo.displayName)
-    # - generate a slug based on display name
-    if not profile.slug:
-        profile.slug = slugify(ldapinfo.displayName)
-    if ldapinfo.telephoneNumber and not profile.phone_number:
-        profile.phone_number = str(ldapinfo.telephoneNumber)
-    # 'street' in ldap is office location
-    if ldapinfo.street and not profile.office_location:
-        profile.office_location = str(ldapinfo.street)
-    # organizational unit = department
-    if ldapinfo.ou and not profile.department:
-        profile.department = str(ldapinfo.ou)
-    # Store job title as string.
-    # NOTE: we may want to split title and only use the first portion.
-    # if ldapinfo.title and not profile.job_title:
-    profile.job_title = str(ldapinfo.title).split('.')[0]
+    # populate Person fields with data we can pull from ldap, if they are empty
+    person, _created = Person.objects.get_or_create(user=user)
 
-    # always update PU status to current
-    profile.pu_status = str(ldapinfo.pustatus)
+    # set first/last name basic on basic whitespace split; can adjust later
+    first_name, *_others, last_name = str(ldapinfo.displayName).split()
+    person.first_name = person.first_name or first_name
+    person.last_name = person.last_name or last_name
 
-    profile.save()
+    # set phone number
+    if ldapinfo.telephoneNumber and not person.phone_number:
+        person.phone_number = str(ldapinfo.telephoneNumber)
 
-    # NOTE: job title is available in LDAP; attaching to a person
-    # currently requires at least a start date (which is not available
-    # in LDAP), but we can at least ensure the title is defined
-    # so it can easily be associated with the person
+    # set office location ("street" in ldap)
+    if ldapinfo.street and not person.office_location:
+        person.office_location = str(ldapinfo.street)
 
-    # only do if the person has a title set
-    if ldapinfo.title:
-        # job title in ldap is currently stored as
-        # job title, organizational unit
-        job_title = str(ldapinfo.title).split(',')[0]
-        Title.objects.get_or_create(title=job_title)
```

Comment:
Are we dropping this? I think I'm ok with that (it ended up generating a lot of extraneous titles we didn't need), just want to be explicit about it. Maybe a comment about previous behavior?

Reply:
Yeah, it seemed like it would cause confusion because `Title`s are largely for positions at CDH, and this would basically create a new one for every possible job title at Princeton (which would explain why there are so many weird ones in the current system). I think setting `job_title` on the `Person` is plenty. I'll add a comment